walkdir.py

```diff
@@ -202,11 +202,32 @@ def onloop(dirpath):
                     continue
         yield dir_entry
 
+
+def handle_dirsymlink_as_file(walk_iter):
+    """Handle symlink to directories as files during the walk.
+
+       By default os.walk insert symlink to directories to the subdirs list.
+       This function move symlink to directories from the subdirs list to the file list.
```

Comment:
I'd reverse the order here, and put the description of the default behaviour in parentheses:

    Move symlinks to directories from the subdirs list to the file list. (By default, os.walk and os.fwalk
    include symlinks to directories in the subdirs list, even when *followlinks* is False)